### PR TITLE
AG-13709 clean changed row nodes nodes added and removed in the same batch

### PR DESCRIPTION
**Pull request by SalvatorePreviti**
As noted in https://github.com/ag-grid/ag-grid/pull/9619#discussion_r1900847905 we can avoid to keep the removed nodes if they were just added and removed in the same transaction batch.
Note that this won't happen with immutable row data but just with asynchronous transaction batches

However - to consider - applying just this change causes a minor change in behaviour compared to latest (and previous ag-grid versions) as the previous version was still calling modelUpdate event for a node added and removed in the same batch - with this change it does not.

From a certain point of view seems more correct that modelUpdated is not called as the transaction batch in this case would result in no changes, but is a different behaviour from previous versions and is a different behaviour if transactions were to be executed one by one that can be a bit confusing.

Todo: decide if we want to maintain the old behaviour or move to this new behaviour.

If we do not want to change the behaviour for this edge case, and apply this change, we need another change to call modelUpdated, a flag for example.

### CODE REVIEW

File: testing/behavioural/src/row-data/row-data.test.ts [hunk 1]
```diff
@@ -183,8 +183,8 @@ describe('ag-grid rows-ordering', () => {
             └── LEAF id:7 value:700
         `);
         expect(rowDataUpdatedCount).toBe(5);
-        expect(modelUpdatedCount).toBe(4);
-        expect(compareCalled).toBe(true);
+        expect(modelUpdatedCount).toBe(3);
```
Comment (SalvatorePreviti) [inline testing/behavioural/src/row-data/row-data.test.ts:186]: note that this is the result of adding and removing node 8 in the same batch - this change results in modelUpdate event to not be raised
